File: server/src/main/scala/producer/message_generator/value_generator/ValueFromTopicSchemaGenerator.scala
package producer.message_generator.value_generator

import com.tools.teal.pulsar.ui.producer.v1.producer as pb

case class ValueFromTopicSchemaGenerator()

object ValueFromTopicSchemaGenerator:
    def fromPb(v: pb.ValueFromTopicSchemaGenerator): ValueFromTopicSchemaGenerator =
        ValueFromTopicSchemaGenerator()

    def toPb(v: ValueFromTopicSchemaGenerator): pb.ValueFromTopicSchemaGenerator =
        pb.ValueFromTopicSchemaGenerator()

File: ui/components/ui/ProducerSession/producer-task/message-generator/value-generator/value-from-topic-schema-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type ValueFromTopicSchemaGenerator = {
  type: 'value-from-topic-schema'
};

export function valueFromTopicSchemaGeneratorFromPb(v: pb.ValueFromTopicSchemaGenerator): ValueFromTopicSchemaGenerator {
  return {
    type: 'value-from-topic-schema'
  };
}

export function valueFromTopicSchemaGeneratorToPb(v: ValueFromTopicSchemaGenerator): pb.ValueFromTopicSchemaGenerator {
  const generatorPb = new pb.ValueFromTopicSchemaGenerator();

  return generatorPb;
}
